File: OOPL/Week8/Q2.java
import java.util.*;
import java.lang.*;

class RowSum implements Runnable
{
	private int arr[];
	private int sum;
	Thread t;

	RowSum(int arr[], String threadName)
	{
		t = new Thread(this,threadName);
		this.arr=arr;
		sum=0;
		t.start();
	}

	public void run()
	{
		for(int i=0;i<arr.length;i++)
			sum=sum+arr[i];
		System.out.println("Sum of "+t.getName()+" = "+sum);
	}
	int getSum()
	{
		return sum;
	}
}

public class Q2
{
	public static void main(String args[])
	{
		Scanner sc = new Scanner(System.in);
		System.out.println("Enter no. of rows and columns : ");
		int r = sc.nextInt();
		int c = sc.nextInt();
		int arr[][] = new int[r][c];
		RowSum t[] = new RowSum[r];
		int sum=0;
		System.out.println("Enter elements : ");
		for(int i=0;i<r;i++)
		{
			for(int j=0;j<c;j++)
				arr[i][j]=sc.nextInt();
		}
		for(int i=0;i<r;i++)
		{
			t[i] = new RowSum(arr[i],"row "+i);
		}
		for(int i=0;i<r;i++)
		{
			try
			{
				t[i].t.join();
			}
			catch(InterruptedException e)
			{
				System.out.println("IE caught");
			}
		}
		for(int i=0;i<r;i++)
		{
			sum = sum + t[i].getSum();
		}
		System.out.println("Total sum = "+sum);
	}
}

File: DS lab/Week 3/q2.py
import numpy as np

r = int(input("Enter the no. of rows: ")) 
c = int(input("Enter the no. of columns: ")) 
matrix = []

print("Enter the elements:") 
for i in range(r):
    a =[] 
    for j in range(c):
         a.append(int(input())) 
    matrix.append(a) 

print("Matrix:")
for i in range(r): 
    for j in range(c): 
        print(matrix[i][j], end = " ") 
    print() 

print("Sum of all columns: ", np.sum(matrix, axis=0))
print("Sum of all rows: ", np.sum(matrix, axis=1))